ako-crd-operator/internal/utils/utils.go

```diff
@@ -306,3 +313,195 @@ func GetTenantInNamespace(ctx context.Context, client client.Client, namespace s
+				select {
+				case annotationFoundCh <- cloudName:
+				default:
+				}
+			}
+		},
+	})
+
+	// Start the informer
+	go informerFactory.Start(stopCh)
+
+	// Wait for informer cache to sync
+	if !cache.WaitForCacheSync(stopCh, nsInformer.Informer().HasSynced) {
+		close(stopCh)
+		utilruntime.HandleError(fmt.Errorf("timed out waiting for namespace informer cache to sync"))
+		return fmt.Errorf("timed out waiting for namespace informer cache to sync")
+	}
+
+	logger.Info("Namespace informer cache synced, waiting for WCP cloud name annotation")
+
+	// Wait for annotation to be found
+	select {
+	case cloudName := <-annotationFoundCh:
+		close(stopCh)
+		logger.Infof("WCP cloud name annotation received: %s", cloudName)
+		return initializeClusterConfig(ctx, kubeClient, cloudName, logger)
+	case <-ctx.Done():
+		close(stopCh)
+		return ctx.Err()
+	}
+}
+
+// initializeClusterConfig retrieves cluster configuration from configmap and validates AVI secret
+func initializeClusterConfig(ctx context.Context, kubeClient kubernetes.Interface, cloudName string, logger *utils.AviLogger) error {
+	logger.Infof("Initializing cluster configuration with cloud name: %s", cloudName)
+
+	// Get avi-k8s-config configmap
+	configMap, err := kubeClient.CoreV1().ConfigMaps(utils.GetAKONamespace()).Get(ctx, "avi-k8s-config", metav1.GetOptions{})
+	if err != nil {
+		logger.Errorf("Failed to get avi-k8s-config configmap: %v", err)
+		return err
+	}
+
+	// Extract clusterID from configmap
+	clusterID := configMap.Data["clusterID"]
+	if clusterID == "" {
+		logger.Error("clusterID not found in avi-k8s-config configmap")
+		return fmt.Errorf("clusterID not found in configmap")
+	}
+	clusterName := clusterID
+	clusterIDArr := strings.Split(clusterID, ":")
+	if len(clusterIDArr) > 1 {
+		// Include first 5 characters to add more uniqueness to cluster name
+		clusterName = clusterIDArr[0] + "-" + clusterIDArr[1][:5]
+	} else {
+		if len(clusterID) > 12 {
+			clusterName = clusterID[:12]
+		}
+	}
+
+	// Extract controllerIP from configmap
+	controllerIP := configMap.Data["controllerIP"]
+	if controllerIP == "" {
+		logger.Error("controllerIP not found in avi-k8s-config configmap")
+		return fmt.Errorf("controllerIP not found in configmap")
+	}
+
+	// Set cluster Name, ID and controller IP
+	os.Setenv("CLUSTER_NAME", clusterName)
+	lib.SetClusterID(clusterID)
+	lib.SetControllerIP(controllerIP)
+	logger.Infof("Set clusterName: %s, clusterID: %s, controllerIP: %s", clusterName, clusterID, controllerIP)
+
+	// Validate AVI secret
+	if err := validateAviSecret(ctx, kubeClient, controllerIP, logger); err != nil {
+		logger.Errorf("AVI secret validation failed: %v", err)
+		return err
+	}
+
+	logger.Info("Cluster configuration initialized successfully")
+	return nil
+}
+
+// validateAviSecret validates the AVI secret by creating an AVI client
+func validateAviSecret(ctx context.Context, kubeClient kubernetes.Interface, controllerIP string, logger *utils.AviLogger) error {
```

Comment:
this function seems to be same as in github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/k8s/ path. Is there a reason for not reusing it?

Reply:
Both are different modules and internal/k8s is not available to ako-crd-op module